modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/UserAccountResourceImpl.java

```diff
@@ -944,23 +944,8 @@ public UserAccount putUserAccount(
 				"Unable to put pending user account " + user.getUserId());
 		}
 
-		String status = userAccount.getStatusAsString();
-
-		Integer workflowStatus = null;
-
-		if (StringUtil.equalsIgnoreCase(
-				UserAccount.Status.ACTIVE.getValue(), status)) {
-
-			workflowStatus = WorkflowConstants.STATUS_APPROVED;
-		}
-		else if (StringUtil.equalsIgnoreCase(
-					UserAccount.Status.INACTIVE.getValue(), status)) {
-
-			workflowStatus = WorkflowConstants.STATUS_INACTIVE;
-		}
-		else {
-			throw new BadRequestException("Status is invalid");
-		}
+		Integer workflowStatus = _getWorkflowStatus(
+			userAccount.getStatusAsString());
```

Comment:
Also I think we need to verify the default status we are using in POST method because:
- in PUT we should return the default status if not passed (now we are throwing an error probably)
- in PATCH we should use the current status if not passed (now we will throw the error and it's wrong because I don't need to pass the status every time I would like to update something)

What do you think?

Reply:
@smottal Currently in Patch we just update the status if it's either "Active" or "Inactive", if it isn't one of these, then we ignore it and no error is thrown ([here](https://github.com/liferay/liferay-portal/blob/master/modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/UserAccountResourceImpl.java#L618-L636)).

In Put we're throwing the exception if the status is invalid but i think that's the expected behavior of PUT, no? If not, what would the default status be?.

And for POST, we can only send "Active" or "Inactive", because of how it's specified in the [yaml](https://github.com/liferay/liferay-portal/blob/ee21e2f265df0d55b2c80ee41d651eb266ac2a01/modules/apps/headless/headless-admin-user/headless-admin-user-impl/rest-openapi.yaml#L1107-L1111). If we send anything other than these, then it's a bad request